bpf/lib/encrypt.h

```diff
@@ -364,6 +364,9 @@ ipsec_maybe_redirect_to_encrypt(struct __ctx_buff *ctx, __be16 proto,
 	if (!dst)
 		return CTX_ACT_OK;
 
+	if (!dst->tunnel_endpoint)
```

Comment:
Okay I'm thinking this is fine.

The sketchy part would be for VXLAN traffic. In v1.17 the IPCache entry for a remote node has 0 as a tunnel_endpoint. 

But looks like that's okay here, since we handle tunnel traffic separately above and jump over this check. 

The only other scenarios where tunnel_endpoint is null then is:
- Dst is World
- Dst is Local 

Any others to consider?

Reply:
Yeah I am just pointing out, in the VXLAN case, the ordering of this code is important.

Reply:
oh yeah, but for VXLAN we're not in this path and we just use the outer `dst_ip` for the node_map lookup. So that's alright.